src/workingtitle-vcockpits-instruments-navsystems-g3000/html_ui/WorkingTitle/Pages/VCockpit/Instruments/NavSystems/AS3000/MFD/AS3000_MFD.js

```diff
@@ -0,0 +1,768 @@
+class AS3000_MFD extends NavSystem {
+    constructor() {
+        super();
+        this.initDuration = 5500;
+        this.needValidationAfterInit = true;
+    }
+    get IsGlassCockpit() { return true; }
+    get templateID() { return "AS3000_MFD"; }
+    connectedCallback() {
+        super.connectedCallback();
+        this.pagesContainer = this.getChildById("RightInfos");
+        this.addIndependentElementContainer(new AS3000_Engine("Engine", "LeftInfos"));
+        this.addIndependentElementContainer(new NavSystemElementContainer("Com Frequencies", "ComFreq", new AS3000_MFD_ComFrequencies()));
+        this.addIndependentElementContainer(new NavSystemElementContainer("Navigation status", "TopRight", new AS3000_MFD_NavInfos()));
+        this.pageGroups = [
+            new NavSystemPageGroup("MAP", this, [
+                new AS3000_MFD_MainMap()
+            ]),
+        ];
+		
+		Include.addScript("/JS/debug.js", function () {
+			g_modDebugMgr.AddConsole(null);
+		});
+    }
+    disconnectedCallback() {
+    }
+    onEvent(_event) {
+        super.onEvent(_event);
+        if (_event == "SOFTKEYS_12") {
+            this.acknowledgeInit();
+        }
+    }
+    Update() {
+        super.Update();
+    }
+}
+
+class AS3000_MFD_WindDataDisplay extends HTMLElement {
```

Comment:
Can you summarize how this differs from the common wind data element?

Reply:
Sure. The graphics are a bit different. Arrow is fatter and shorter. The wind strength display now says "KT" after the number. It also will hide the arrow if wind speed is <1 kt. It will also show NO WIND DATA if the plane's true ground speed is less than 5 knots. I tried to make it as close to the G3000's actual MFD wind vector display as possible. I think I got it pretty close; the only dodgy bit is the no wind data <5 kt thing. I couldn't find a description anywhere of when exactly the G3000 does and does not report wind data. The TBM G3000 manual just says it won't do it when the aircraft is "not moving", which is very vague.

Reply:
Interesting.   The fix we put in CommonPFD_MFD.js for the shared wind bug just checks to see whether the plane is on the ground.    The question of what happens on the takeoff run is interesting.   I just fired up my G1000 trainer, and at least in that I could be doing 200kts on the ground and it still said NO WIND DATA.

Not sure how the 3000 should be have, though.

Reply:
Yeah, I actually initially also did a ON GROUND check. But I kept reading that one sentence in the manual over and over again that says no wind data when not moving and I figured "well technically you can be moving and still be on the ground..." But I don't know what the right answer is either.